src/item/CustomiesItemFactory.php

```diff
@@ -116,4 +145,47 @@ public function registerBlockItem(string $identifier, int $blockId): void {
 		$this->registerCustomItemMapping($itemId);
 		$this->itemTableEntries[] = new ItemTypeEntry($identifier, $itemId, false);
 	}
+
+    /**
+     * getNextAvailableId retrieves the lowest unregistered item id above 950. It will check the item id cache if caching is
+     * enabled.
+     */
+	private function getNextAvailableId(string $identifier): int {
```

Comment:
You could clean this but up instead of having a duplicate function both in item factory and block factory.